src/Serilog.Sinks.Datadog.Logs/Serilog.Sinks.Datadog.Logs.csproj

```diff
@@ -26,13 +26,17 @@
     <Compile Remove="Configuration\Implementations\Microsoft.Extensions.Configuration\**\*.*" />
   </ItemGroup>
 
-  <ItemGroup Condition="'$(TargetFramework)' == 'netstandard2.0'">
+  <ItemGroup Condition="'$(TargetFramework)' == 'netstandard2.0' Or '$(TargetFramework)' == 'net472'">
     <PackageReference Include="Microsoft.Extensions.Options.ConfigurationExtensions" Version="2.1.1" />
```

Comment:
Microsoft.Extensions.Options.ConfigurationExtensions requiers .NET Standard 2.0 and the minimal compatible version of .NET Framework is 4.6.1: I think that you can downgrade the new target framework to net461 instead of net472.

Reply:
see https://docs.microsoft.com/en-us/dotnet/standard/net-standard#net-implementation-support : 

> 2 The versions listed here represent the rules that NuGet uses to determine whether a given .NET Standard library is applicable. While NuGet considers .NET Framework 4.6.1 as supporting .NET Standard 1.5 through 2.0, there are several issues with consuming .NET Standard libraries that were built for those versions from .NET Framework 4.6.1 projects. For .NET Framework projects that need to use such libraries, we recommend that you upgrade the project to target .NET Framework 4.7.2 or higher.

Reply:
I've personally had an infinite number of issues with pre 4.7.2 framework projects and standard support, as Microsoft suggets, hence why i chose to set it at 4.7.2 and not 4.8, for example. It's the first Framework version that really support .NET Standard 2.0 properly.

Reply:
The issue you describe in #39 would happen to .NET 4.6.1-4.7.1 users as well? (I verified that locally and it seems to be the case).

I think @sjvl-smart suggestion would be better, as a project targeting >= 4.7.2 would select the 461 version of this library and should get the correct dependencies? Or another option is that you target both 461 and 472 and have conditional for both. 

I agree that 4.6.1 supporting .NET Standard 2.0 is a royal pain to go through (we are finally at .NET 4.8 for all projects not already migrated to .NET Core), but in this case it should not be an issue targeting 4.6.1 for the serilog-sink?

Reply:
If maintaining NET 4.6.1-4.7.1 compatibility is the case then id take the latter option you mention.

At this point, supporting anything below 4.7.2 with netstandard compatible versions is not only a royal pain, its simply not really supported - not by the tooling, not by NuGet.
Lowering the target WILL cause issues, guaranteed. (by MS!)

Reply:
There is support for .NET 4.5, so I guess that 4.6.1-4.7.1 should also be supported (otherwise 4.5 could also be dropped)?

Reply:
There's probably an argument that anything below 4.7.2 can just use the netstandard1.3 target that's already there unless the net45 target was added exactly because of incompatibility with that.

Reply:
The NuGet resolving algorithm (if I understand it correctly) would always pick the most specific target. So .NET 4.6.1 would always pick net45 over netstandard2.0 if that exists unfortunately, so to support .NET 4.6.1-4.7.1, the net461 target should also be added I guess..

Reply:
@ransagy :
>Lowering the target WILL cause issues, guaranteed. (by MS!)

Even if it is recommended to upgrade to `.NET Framework 4.7.2` when working with `.NET Standard 2`, is there a reason to not support `4.6.1` (Replace `4.7.2` by `4.6.1` in this PR)? 
Is there a case where a user using `.NET Framework 4.7.2`  has an issue if in this PR, `4.7.2` is replaced by `4.6.1` (If yes which one)?
Are you saying that trying to support a `NET Framework 4.6.1` project using a `.NET Standard 2` project is something that do not need to be considered as it is not working in practice?

Reply:
That is my understand, yes; That as long as a target lower than 4.7.2/netstandard2.0 is there, it will cause issues for higher targets if thats the most compatible one - as NuGet/tooling will make the wrong choices for some inbox assemblies and such, for anything between 4.6.1 (where they started the move of taking inbox assemblies to NuGets) and 4.7.2 (where they fully reverted said move and correctly fixed all the tooling and how NuGet selects assemblies for those targets).

Reply:
Can you also offer a `net461` build and keep `4.7.2` build?

Reply:
Just pushed a rebase with changes to that effect, feel free to re-review @ogaca-dd @pergardebrink @sjvl-smart